nautobot/extras/templates/extras/job.html

```diff
@@ -70,3 +95,52 @@ <h1>{{ job }}</h1>
         </div>
     </div>
 {% endblock %}
+{% block javascript %}
+<script type="text/javascript">
+    var requires_approval = {{ job.approval_required|yesno:"true,false" }};
+    var run_now_text;
+    var schedule_text;
+
+    if (requires_approval) {
+        run_now_text = "Request to Run Job Now";
+        schedule_text = "Request to Schedule Job";
+    } else {
+        run_now_text = "Run Job Now";
+        schedule_text = "Schedule Job";
+    }
+
+    function toggleExecutionType() {
+        /**
+         * Toggle form elements based on the job execution type.
+         * If the type is `immediately` (the default), don't show
+         * any of the scheduling related elements, else, show them.
+         **/
+        if ($("#id__schedule_type").val() == "immediately") {
+            // Toggle run button
+            $("#id__run").html('<i class="mdi mdi-play"></i> ' + run_now_text);
+            $("#id__run").addClass("btn-primary");
+            $("#id__run").removeClass("btn-warning");
+
+            // Toggle schedule fields
+            $("#id__schedule_name").parent().parent().hide();
+            $("#id__schedule_start_date").parent().parent().hide();
+            $("#id__schedule_start_time").parent().parent().hide();
+        } else {
```

Comment:
Is there a common parent of all three of these fields that we could assign an id to so as to hide/show it directly in one step?

Reply:
Unfortunately, these are three fields (rendered as two) in a form which are hidden/shown dependent on another field in that same form. So their common ancestor is the form itself, which needs to be shown at all times, and we defer the rendering to Django in line 73.

We could render the form manually instead, but that would incur that cost for saving a few lines. I’d opt for this instead. What do you think?

Reply:
Good enough for now I think. Thanks for investigating!